### ISSUE TITLE
tf.cumsum low performance in 1.12.0

### ISSUE THREAD

**Issue by HansenZhao** [tensorflow/tensorflow#24362]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 1803 Build 17134.407
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.12.0-gpu
- Python version: 3.6.0
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: v9.0.176
- GPU model and memory: GTX-970M, NVIDIA-SMI 385.54

**Describe the current behavior**
In tensorflow-1.12.0-GPU, it takes around 0.8s for each iteration and the cumsum operation takes around 0.13s/run x 3class x 2run/class. So the cumsum operation nearly takes 0.78s for each iteration. After switching the code interpreter to python 3.6.0 with tensorflow-1.11.0-CPU, the iteration time decreases to around 0.14s and the highest time-consuming operation become the sorting operation.
I notice that this issue has been reported in [https://github.com/tensorflow/tensorflow/issues/19570](url). I wonder whether this issue has been fixed in version 1.12.0? Or is there something wrong in my code caused this issue? I'm still learning how to use tensorflow recently. The lovasz_loss imported can be found in [https://github.com/bermanmaxim/LovaszSoftmax/blob/master/tensorflow/lovasz_losses_tf.py](url)

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
`
import tensorflow as tf
import numpy as np
import time
import lovasz_loss
from tensorflow.python.client import timeline

print(tf.__version__)

IMAGE_SIZE = 512
IMAGE_CHANNEL = 2
CLASS_NUM = 3
IMAGE_NUM = 1
X_train = np.random.randn(IMAGE_NUM,IMAGE_SIZE,IMAGE_SIZE,IMAGE_CHANNEL)
y_train = np.random.randint(0,CLASS_NUM,size=[IMAGE_NUM,IMAGE_SIZE,IMAGE_SIZE],dtype=np.int32)

X = tf.placeholder(dtype=tf.float32,shape=[None,IMAGE_SIZE,IMAGE_SIZE,IMAGE_CHANNEL],name='image_input')
y = tf.placeholder(dtype=tf.int32,shape=[None,IMAGE_SIZE,IMAGE_SIZE],name='image_output')

w = tf.get_variable('w',initializer=tf.truncated_normal(shape=[1,1,IMAGE_CHANNEL,CLASS_NUM],stddev=0.02))
y_logits = tf.nn.conv2d(X,w,strides=[1,1,1,1],padding='SAME')
y_prob = tf.nn.softmax(y_logits,axis=3)

loss = lovasz_loss.lovasz_softmax(y_prob,y)
train_op = tf.train.AdamOptimizer(0.005).minimize(loss)

config = tf.ConfigProto()
config.gpu_options.allow_growth = True

run_option = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
run_metadata = tf.RunMetadata()

with tf.Session(config=config) as sess:
    tf.global_variables_initializer().run()
    tf.local_variables_initializer().run()
    for i in range(100):
        st = time.time()
        sess.run(train_op,feed_dict={X:X_train,y:y_train},options=run_option,run_metadata=run_metadata)
        if i>5:
            tl = timeline.Timeline(run_metadata.step_stats)
            ctf = tl.generate_chrome_trace_format()
            with open('timeline_lova.json','w') as f:
                f.write(ctf)
        print('Iter: {num}, duration: {sec}'.format(num=i,sec=time.time()-st))
`
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by sanjoy** [comment]
Closing out this issue because it hasn't been updated in the last year.  Please reopen if this is still relevant.
